From: Chris Liddell <chris.liddell@artifex.com>
Date: Fri, 6 Dec 2019 21:45:09 +0000 (+0000)
Subject: Bug 701973: Fix gssprintf.c includes for PRI macros
X-Git-Url: https://git.ghostscript.com/?p=ghostpdl.git;a=commitdiff_plain;h=b46142c94bde

Bug 701973: Fix gssprintf.c includes for PRI macros

The header inclusion in gssprintf.c meant the definitions of the PRI macros
for string formatting could be inconsistent with other source files.

Adding the appropriate headers in the correct order resolves that issue.
---

diff --git a/base/gssprintf.c b/base/gssprintf.c
index 49f54bd..aab4051 100644
--- a/base/gssprintf.c
+++ b/base/gssprintf.c
@@ -16,8 +16,9 @@
 /* From: https://svn.apache.org/repos/asf/apr/apr/trunk/strings/apr_snprintf.c */
 /* Additional modifications to work outside Apache for use with Ghostscript */
 
+#include "unistd_.h"
+#include "gp.h"
 #include "stdio_.h"
-#include "stdint_.h"
 #include "string_.h"
 #include <stdarg.h>
 #include "math_.h"
diff --git a/base/lib.mak b/base/lib.mak
index 534e2dc..7e8f431 100644
--- a/base/lib.mak
+++ b/base/lib.mak
@@ -382,8 +382,8 @@ $(AUX)gsutil.$(OBJ) : $(GLSRC)gsutil.c $(AK) $(memory__h) $(string__h)\
  $(gsrect_h) $(gsuid_h) $(gsutil_h) $(LIB_MAK) $(MAKEDIRS)
 	$(GLCCAUX) $(C_) $(AUXO_)gsutil.$(OBJ) $(GLSRC)gsutil.c
 
-$(GLOBJ)gssprintf.$(OBJ) : $(GLSRC)gssprintf.c $(gssprintf_h) $(stdio__h) \
- $(stdint__h) $(string__h) $(math__h)
+$(GLOBJ)gssprintf.$(OBJ) : $(GLSRC)gssprintf.c $(gssprintf_h) $(unistd__h) \
+ $(gp_h) $(stdio__h) $(stdint__h) $(string__h) $(math__h)
 	$(GLCC) $(GLO_)gssprintf.$(OBJ) $(C_) $(GLSRC)gssprintf.c
 
 $(GLOBJ)gsstrtok.$(OBJ) : $(GLSRC)gsstrtok.c $(gsstrtok_h) $(string__h)
